viam_cartographer.go

```diff
@@ -463,13 +466,13 @@ func terminateCartoFacade(ctx context.Context, cartoSvc *CartographerService) er
 type CartographerService struct {
 	resource.Named
 	resource.AlwaysRebuild
-	mu                 sync.Mutex
-	SlamMode           cartofacade.SlamMode
-	closed             bool
-	lidar              s.TimedLidar
-	movementSensor     s.TimedMovementSensor
-	movementSensorName string
-	subAlgo            SubAlgo
+	SlamMode       cartofacade.SlamMode
+	lidar          s.TimedLidar
+	movementSensor s.TimedMovementSensor
+	subAlgo        SubAlgo
+
+	muClose sync.Mutex
+	closed  bool
```

Comment:
Note Rand, This came up in an eng talk by Tess six months back. Someone told me that the mod manager should be handling this now, but when I talked to Benji, he had no idea what I was talking about and didn't know if it was being handled so we decided to keep this closed check in, until we know for certain.

Reply:
There was a slakc thread about this, it's not mod manager, it's the resource graph that handles removing the resource nodes from the graph (modules included). Its not something other modules do, since if it's closed the client only returns an error for a resource that's not there anymore.

Like if you try to import the resource from the robot after it's closed and not built again, the `FromRobot`, `FromDep` call in your code just fails. If it doesn't, that's a netcode bug.

Reply:
Great! Then we don't need the `closed` or `closedMu`! @randhid should we handle that in this PR or create a ticket to be solved now (this PR is already pretty big, but based on that, the only change would need to be code removal)

Reply:
So the only case to keep this is if cloud slam is directly importing cartographer methods outside of the resource graph.

For example, `gpsnmea` has public functions that are imported directly from within rdk by `gpstrk` (this is something we want to fix but never have bandwidth to), so we have a closed bool there to account for one structure being imported directly outside of the resource graph logic. 

If nothing is doing so with this module then it's unnecessary, but I want to be exhaustive about the potential gotchas.

Reply:
Kat, have you seen any of these errors or logs surface at all? Is the C++ layer taking a long time to shut down and thus we are in danger of calling an endpoint directly before the resource graph removes this resource node?

Reply:
I have neither paid attention nor noticed it. Can we create a separate ticket to look into this?

Reply:
Then let's not add code that might not do anything at all and wasn't hurting the implementation, but does seem to be important for anyone reading the code.

Reply:
I was referring to your mutexes, okay to remove the closed checks to another pr.